discretize/base/base_tensor_mesh.py

```diff
@@ -341,62 +364,69 @@ def _getInterpolationMat(self, loc, locType='CC', zerosOutside=False):
-            assert self.dim >= ind, 'mesh is not high enough dimension.'
-            nF_nE = self.vnF if 'F' in locType else self.vnE
-            components = [utils.spzeros(loc.shape[0], n) for n in nF_nE]
-            components[ind] = utils.interpmat(loc, *self.getTensor(locType))
+            indZeros = np.logical_not(self.is_inside(loc))
+            loc[indZeros, :] = np.array([v.mean() for v in self.get_tensor("CC")])
+
+        if loc_type in ["Fx", "Fy", "Fz", "Ex", "Ey", "Ez"]:
+            ind = {"x": 0, "y": 1, "z": 2}[loc_type[1]]
+            if self.dim < ind:
+                raise ValueError("mesh is not high enough dimension.")
+            if "F" in loc_type:
+                items = (self.nFx, self.nFy, self.nFz)[: self.dim]
+            else:
+                items = (self.nEx, self.nEy, self.nEz)[: self.dim]
+            components = [spzeros(loc.shape[0], n) for n in items]
+            components[ind] = interpolation_matrix(loc, *self.get_tensor(loc_type))
             # remove any zero blocks (hstack complains)
             components = [comp for comp in components if comp.shape[1] > 0]
             Q = sp.hstack(components)
 
-        elif locType in ['CC', 'N']:
-            Q = utils.interpmat(loc, *self.getTensor(locType))
+        elif loc_type in ["CC", "N"]:
+            Q = interpolation_matrix(loc, *self.get_tensor(loc_type))
 
-        elif locType in ['CCVx', 'CCVy', 'CCVz']:
-            Q = utils.interpmat(loc, *self.getTensor('CC'))
-            Z = utils.spzeros(loc.shape[0], self.nC)
-            if locType == 'CCVx':
+        elif loc_type in ["CCVx", "CCVy", "CCVz"]:
+            Q = interpolation_matrix(loc, *self.get_tensor("CC"))
+            Z = spzeros(loc.shape[0], self.nC)
+            if loc_type == "CCVx":
                 Q = sp.hstack([Q, Z, Z])
-            elif locType == 'CCVy':
+            elif loc_type == "CCVy":
                 Q = sp.hstack([Z, Q, Z])
-            elif locType == 'CCVz':
+            elif loc_type == "CCVz":
                 Q = sp.hstack([Z, Z, Q])
 
         else:
             raise NotImplementedError(
-                'getInterpolationMat: locType==' + locType +
-                ' and mesh.dim==' + str(self.dim)
+                "getInterpolationMat: loc_type=="
+                + loc_type
+                + " and mesh.dim=="
+                + str(self.dim)
             )
 
-        if zerosOutside:
+        if zeros_outside:
             Q[indZeros, :] = 0
 
         return Q.tocsr()
 
-    def getInterpolationMat(self, loc, locType='CC', zerosOutside=False):
-        """ Produces interpolation matrix
+    def get_interpolation_matrix(
+        self, loc, loc_type="CC", zeros_outside=False, **kwargs
+    ):
+        """Produces linear interpolation matrix
 
         Parameters
         ----------
         loc : numpy.ndarray
             Location of points to interpolate to
 
-        locType : str
+        loc_type : str
```

Comment:
would it be worth updating this to `location_type` to be a bit more explicit (since we are renaming anyways)